repos/spack_repo/builtin/packages/gcc/package.py

```diff
@@ -993,6 +1076,86 @@ def nvptx_install(self):
             make()
             make("install")
 
+    # run configure/make/make(install) for the amdgcn-amdhsa target
+    # before running the host compiler phases
+    # See https://gcc.gnu.org/wiki/Offloading#For_AMD_GCN:
+    @run_before("configure")
+    def amdgcn_install(self):
+        spec = self.spec
+        prefix = self.prefix
+
+        if not spec.satisfies("+amdgcn"):
+            return
+
+        # config.guess returns the host triple, e.g. "x86_64-pc-linux-gnu"
+        guess = Executable("./config.guess")
+        targetguess = guess(output=str).rstrip("\n")
+
+        options = getattr(self, "configure_flag_args", [])
+        options += [f"--prefix={prefix}"]
+
+        # Copy LLVM utils
+        llvm_bin_path = join_path(spec["llvm"].prefix, "bin")
+        llvm_util_path = join_path(prefix, "amdgcn-amdhsa", "bin")
+        mkdirp(llvm_util_path)
+        copy(
+            "{0}".format(join_path(llvm_bin_path, "llvm-ar")),
+            "{0}".format(join_path(llvm_util_path, "ar")),
+        )
+        copy(
+            "{0}".format(join_path(llvm_bin_path, "llvm-ar")),
+            "{0}".format(join_path(llvm_util_path, "ranlib")),
+        )
+        copy(
+            "{0}".format(join_path(llvm_bin_path, "llvm-mc")),
+            "{0}".format(join_path(llvm_util_path, "as")),
+        )
+        copy(
+            "{0}".format(join_path(llvm_bin_path, "llvm-nm")),
+            "{0}".format(join_path(llvm_util_path, "nm")),
+        )
+        copy(
+            "{0}".format(join_path(llvm_bin_path, "lld")),
+            "{0}".format(join_path(llvm_util_path, "ld")),
+        )
```

Comment:
This part doesn't work. Inspecting the install tree for GCC you can see these files have not been copied.